modules/mbedtls/crypto_mbedtls.cpp

```diff
@@ -186,6 +188,78 @@ Error X509CertificateMbedTLS::save(String p_path) {
 	return OK;
 }
 
+bool HMACContextMbedTLS::is_md_type_allowed(mbedtls_md_type_t p_md_type) {
+	mbedtls_md_type_t allowed[] = { MBEDTLS_MD_SHA1, MBEDTLS_MD_SHA256 };
+
+	for (size_t i = 0; i < sizeof(allowed); i++) {
+		if (p_md_type == allowed[i]) {
+			return true;
+		}
+	}
+	return false;
+}
+
+HMACContext *HMACContextMbedTLS::create() {
+	return memnew(HMACContextMbedTLS);
+}
+
+HMACContextMbedTLS::HMACContextMbedTLS() {
+	ctx = memalloc(sizeof(mbedtls_md_context_t));
+	mbedtls_md_init((mbedtls_md_context_t *)ctx);
+}
+
+HMACContextMbedTLS::~HMACContextMbedTLS() {
+	mbedtls_md_free((mbedtls_md_context_t *)ctx);
+	memfree((mbedtls_md_context_t *)ctx);
+	ctx = nullptr;
+}
+
+Error HMACContextMbedTLS::start(HashingContext::HashType p_hash_type, PackedByteArray p_key) {
+	ERR_FAIL_COND_V_MSG(ctx != nullptr, ERR_FILE_ALREADY_IN_USE, "HMACContext already started.");
```

Comment:
Wouldn't this always fail since `ctx` is assigned in the constructor, and only freed in the deconstructor?
This looks like a bug.

Reply:
Good catch. I actually meant to add unit tests for this, but completely forgot. Ill fix this and add those too.

Reply:
I fixed this and added some unit tests for posterity.